analysis/summaries/frontend_test.go

```
@@ -0,0 +1,53 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package summaries
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestParseSummariesFile(t *testing.T) {
+	summaries, err := ParseSummariesFile("testdata/summaries.yaml")
+	if err != nil {
+		t.Error(err)
+	}
+	t.Logf("Summaries: %+v", summaries)
+}
+
+func TestParseSummariesInvalidFile(t *testing.T) {
+	_, err := ParseSummariesFile("testdata/summaries-invalid.yaml")
+	if err == nil {
+		t.Error("Expected error")
+	}
+	expectedSubstrs := []string{
```

nit: testing error strings is error-prone and will be painful to maintain if we change the error messages in the future. Instead, you can return a public error type like `ErrEmptyPackage` from `ParseSummariesFile` and then use `errors.Is` in the test.

---

In this case I'm testing the user interface, I want to avoid someone to inadvertently change the error messages that should be surfaced to the user. I can add a comment in the tests. I don't want to add the error types just for testing; maybe if we need more error handling in the non-testing code then yes.